core/rust/utils/src/account.rs

```diff
@@ -65,17 +65,28 @@ pub fn resize_or_reallocate_account_raw<'a>(
 ) -> ProgramResult {
     let rent = Rent::get()?;
     let new_minimum_balance = rent.minimum_balance(new_size);
+    let current_ta_lamports = target_account.lamports();
+    let account_infos = &[
+        funding_account.clone(),
+        target_account.clone(),
+        system_program.clone(),
+    ];
 
-    let lamports_diff = new_minimum_balance.saturating_sub(target_account.lamports());
-    invoke(
-        &system_instruction::transfer(funding_account.key, target_account.key, lamports_diff),
-        &[
-            funding_account.clone(),
-            target_account.clone(),
-            system_program.clone(),
-        ],
-    )?;
-
+    // account will be shrunk
+    if target_account.data_len() > new_size {
+        let lamports_diff = new_minimum_balance.saturating_sub(current_ta_lamports);
```

Comment:
I think this if statement is reversed.  Example: target account data len is 200 bytes.  New size is 20 bytes.
New minimum balance will be Rent(20).  Current lamports is Rent(200) which will be bigger than Rent(20).  Therefore lamports diff would be 0.

Looks to me like L84-L88 was meant to be in the `if` statement instead of the `else`.

Reply:
Good catch. Idk how I missed that.